public/pages/AnomalyCharts/utils/anomalyChartUtils.ts

```diff
@@ -138,3 +150,281 @@ export const disabledHistoryAnnotations = (
     },
   ];
 };
+
+export const ANOMALY_HEATMAP_COLORSCALE = [
+  [0, '#F2F2F2'],
+  [0.2, '#F2F2F2'],
+  [0.2, '#F7E0B8'],
+  [0.4, '#F7E0B8'],
+  [0.4, '#F2C596'],
+  [0.6, '#F2C596'],
+  [0.6, '#ECA976'],
+  [0.8, '#ECA976'],
+  [0.8, '#E78D5B'],
+  [1, '#E8664C'],
+];
+
+export enum AnomalyHeatmapSortType {
+  SEVERITY = 'by_severity',
+  OCCURRENCES = 'by_occurrences',
+}
+
+const getColorForValue = (value: number) => {
+  if (
+    value >=
+    ANOMALY_HEATMAP_COLORSCALE[ANOMALY_HEATMAP_COLORSCALE.length - 1][0]
+  ) {
+    return ANOMALY_HEATMAP_COLORSCALE[ANOMALY_HEATMAP_COLORSCALE.length - 1][1];
+  }
+
+  if (value <= ANOMALY_HEATMAP_COLORSCALE[0][0]) {
+    return ANOMALY_HEATMAP_COLORSCALE[0][1];
+  }
+
+  for (let i = 0; i < ANOMALY_HEATMAP_COLORSCALE.length - 1; i++) {
+    if (
+      value >= ANOMALY_HEATMAP_COLORSCALE[i][0] &&
+      value < ANOMALY_HEATMAP_COLORSCALE[i + 1][0]
+    ) {
+      return ANOMALY_HEATMAP_COLORSCALE[i + 1][1];
+    }
+  }
+};
+
+const NUM_CELLS = 20;
+
+export const HEATMAP_X_AXIS_DATE_FORMAT = 'MM-DD HH:mm YYYY';
```

Comment:
minor: was this format specified by UX? Seems kind of confusing, and wondering if we can change to make it consistent with the other date formats on the page, in the form of 'MM/DD/YY h:mm A'

Reply:
Good point. This is actually more for the purpose of getting date range interval, with no `Year` info, there may be issue when [calculating time difference between 2 neighbor timestamps](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/pull/314/files#diff-a044985587be3c93885ad08b0bb90396e88600895b94ace004c30cda74f955d7R233-R234) in 1 special case, where previous timestamp is 12/31, 23:59, and the next one is 01/01, 00:01. In that case, time difference from 01/01, 00:01 to 12/31, 23:59 is totally different from 12/31, 23:59 to 01/01, 00:01. I think I will just add time interval as 1 attribute of PlotData so that we can get rid of `getHeatmapCellDateRangeInterval()`

Reply:
Sorry, not sure I understand. Wouldn't adding the year in the format of 'MM/DD/YY h:mm A' solve this, by comparing against the year as well? To prevent issues with that edge case you mentioned above?

Reply:
sorry it is confusing. Actually format in current single entity anomaly chart has no year in X axis. Using 'MM/DD/YY h:mm A' may not keep it consistent. But adding 1 more attribute solves the problem in a clean way. will push new commit after addressing all comments.